### DISCUSSION TITLE
能否在 xmake.lua 或 xmake.conf 中配置生成 compile_commands.json

### DISCUSSION THREAD

**Discussion by zchrissirhcz** [xmake-io/xmake#2834] [Q&A] [answered]
compile_commands.json 搭配 clangd 能够让 vscode 里的 c++ 代码正确跳转、不显示“下画波浪线”。
比较习惯使用命令行进行编译连接等步骤，不太想尝试 xmake 的 vscode 插件。

目前知道的生成 compile_commands.json 的命令：
```
xmake project -k compile_commands -a outputdir ./build
```

而如果是 CMake, 可以在 `CMakeLists.txt` 中设置 `set(CMAKE_EXPORT_COMPILE_COMMANDS ON)`, 或设置环境变量 `export CMAKE_EXPORT_COMPILE_COMMANDS=1`， 使得默认生成 `compile_commands.json` 文件。

能否在 xmake.lua 或 xmake.conf 中配置生成 compile_commands.json 呢？即：对于新建的c/c++工程，希望只需要输入
```bash
xmake build
xmake run
```

**Comment by waruqi** [comment, marked as answer]
```lua
target("test")
    set_kind("binary")
    add_files("src/*.cpp")
    after_build(function (target)
        import("core.base.task")
        task.run("project", {kind = "compile_commands", outputdir = "build"})
    end)
```

**Reply by zchrissirhcz** [reply]
Cool!
